### DISCUSSION TITLE
Federated Certificate ID - stream id uniqueness

### DISCUSSION THREAD

**Discussion by MartinSchmidt** [project-origin/.github#13] [Documentation issues]
A [Federated Certficate ID (**FID**)] (https://project-origin.github.io/registry/concept/granular-certificates/federated-certifate-id.html) is made up of two parts, the RegistryID and the StreamID, i got a question whether or not the StreamID should be unique only within a single registry, or across the entire federated setup. 

System wise within a registry is enough, since we always use FID when any commands and queries are made. But is there a benefit to enforce it across all registries?

Should this be a question for ETT?

**Comment by MartinSchmidt** [comment]
@wisbech  any comment?

**Reply by wisbech** [reply]
A DID decentralized ID is well described by [W3C](https://www.w3.org/TR/did-core/) as a 3 part id `DID:METHOD:SPECIFIC-IDENTIFIER`, so in a sense it is two parts `DID:registry:registry-specific-id` where `DID`simply specifies that you are dealing with a decentralized id.
